database/opencypher/src/main/java/com/neueda/jetbrains/plugin/graphdb/database/opencypher/gremlin/exceptions/ExceptionWrapper.java

```diff
@@ -0,0 +1,75 @@
+package com.neueda.jetbrains.plugin.graphdb.database.opencypher.gremlin.exceptions;
+
+import org.apache.tinkerpop.gremlin.driver.exception.ConnectionException;
+import org.apache.tinkerpop.gremlin.driver.exception.ResponseException;
+import org.apache.tinkerpop.gremlin.driver.ser.SerializationException;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+
+public class ExceptionWrapper {
+    public static final int SHORT_STRING_LENGTH = 150;
+    private static final String NON_THIN_CHARS = "[^iIl1\\.,']";
+
+    private static int textWidth(String str) {
+        return str.length() - str.replaceAll(NON_THIN_CHARS, "").length() / 2;
+    }
+
+    public static String ellipseString(String text, int targetLength) {
+        if (textWidth(text) <= targetLength) {
+            return text;
+        }
+        int end = text.lastIndexOf(' ', targetLength - 3);
+        if (end == -1) {
+            return text.substring(0, targetLength - 3) + "...";
+        }
+        int newEnd = end;
+        do {
+            end = newEnd;
+            newEnd = text.indexOf(' ', end + 1);
+            if (newEnd == -1) {
+                newEnd = text.length();
+            }
+
+        } while (textWidth(text.substring(0, newEnd) + "...") < targetLength);
+
+        return text.substring(0, end) + "...";
+    }
+
+    public static String getCause(Exception exception) {
+        StringBuilder exceptionCauses = new StringBuilder();
+        Throwable cause = exception.getCause();
+        while (cause != null) {
```

Comment:
If the stacktrace recursively repeats, this can cause an endless loop, hence a StackOverflow.
One solution may be using Apache's lang3 ExceptionUtils.getThrowableList, or implementing the same behavior - put causes in a list and break the loop when trying to add an already existing cause